RockLib.Serialization/DefaultXmlSerializer.cs

```diff
@@ -35,89 +39,121 @@ public DefaultXmlSerializer(string name = "default", XmlQualifiedName[] namespac
-                return new XmlSerializer(type).Deserialize(xmlReader);
+            using var xmlReader = XmlReader.Create(stream, ReaderSettings);
+            return new XmlSerializer(type).Deserialize(xmlReader);
         }
 
         /// <inheritdoc />
         public string SerializeToString(object item, Type type)
         {
-            if (item == null)
+            if (item is null)
+            {
                 throw new ArgumentNullException(nameof(item));
-            if (type == null)
+            }
+            if (type is null)
+            {
                 throw new ArgumentNullException(nameof(type));
+            }
 
             type = CheckType(type, item);
 
             var builder = new StringBuilder();
-            if (WriterSettings == null)
-                using (var writer = new EncodedStringWriter(builder))
-                    new XmlSerializer(type).Serialize(writer, item, Namespaces);
+            if (WriterSettings is null)
+            {
+                using var writer = new EncodedStringWriter(builder);
+                new XmlSerializer(type).Serialize(writer, item, Namespaces);
+            }
             else
-                using (var stringWriter = new EncodedStringWriter(builder, WriterSettings.Encoding))
-                using (var writer = XmlWriter.Create(stringWriter, WriterSettings))
-                    new XmlSerializer(type).Serialize(writer, item, Namespaces);
+            {
+                using var stringWriter = new EncodedStringWriter(builder, WriterSettings.Encoding);
+                using var writer = XmlWriter.Create(stringWriter, WriterSettings);
+                new XmlSerializer(type).Serialize(writer, item, Namespaces);
+            }
 
             return builder.ToString();
         }
 
         /// <inheritdoc />
-        public object DeserializeFromString(string data, Type type)
+        public object? DeserializeFromString(string data, Type type)
         {
-            if (data == null)
+            if (data is null)
+            {
                 throw new ArgumentNullException(nameof(data));
-            if (type == null)
+            }
+            if (type is null)
+            {
                 throw new ArgumentNullException(nameof(type));
-
-            if (ReaderSettings == null)
-                using (var reader = new StringReader(data))
-                    return new XmlSerializer(type).Deserialize(reader);
-
-            using (var reader = new StringReader(data))
-            using (var xmlReader = XmlReader.Create(reader, ReaderSettings))
-                return new XmlSerializer(type).Deserialize(xmlReader);
+            }
+
+            if (ReaderSettings is null)
+            {
+                using var stringReader = new StringReader(data);
+                using var xmlNullSettingsReader = XmlReader.Create(stringReader);
+                return new XmlSerializer(type).Deserialize(xmlNullSettingsReader);
```

Comment:
Previously this Deserialize was using just the StringReader as its parameter.  Will this change the functionality?

Reply:
It doesn't seem to. This was flagged by one of the analyzers - the details are [here](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca5369). All of the tests pass, so....this seems to be fine.